### PR TITLE
Allow poetry in tox allowed list

### PR DESCRIPTION
**Pull request by flaviuvadan**
CC: @samj1912

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [tox]
 envlist = py{37,38,39,310},coverage,lint,typecheck,format
 isolated_build = true
+allowlist_externals = *
```
Comment (samj1912) [inline tox.ini:4]:
```suggestion
allowlist_externals = poetry
```
should this just be poetry?

Comment (flaviuvadan) [reply tox.ini:4]: Tried in https://github.com/argoproj-labs/hera-workflows/pull/389/commits/f6f80aef9a5fe0061291f683c21b8f834a8f4bd8 and it did not seem to work, which is why I wanted to give "*" a go 🤔

**Comment by samj1912** [comment]
@flaviuvadan it seems to me that the issue is the release of tox 4.0 recently. If we pin back to 3.x the old config should work.

**Comment by samj1912** [comment]
See https://github.com/argoproj-labs/hera-workflows/pull/391 should probably work

**Pull request closed without merging**